spec/routemaster/integration/cache_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+require 'spec/support/uses_redis'
+require 'spec/support/uses_dotenv'
+require 'routemaster/cache'
+require 'webrick'
+
+RSpec.describe 'Requests with caching' do
+  uses_dotenv
+  uses_redis
+
+  let(:service) do
+    WEBrick::HTTPServer.new(:Port => 8000, :DocumentRoot => Dir.pwd).tap do |server|
+      server.mount_proc '/test' do |req, res|
+        res.body = { field: 'test' }.to_json
+      end
+    end
+  end
+
+  before do
+    @pid = fork do
+      trap 'INT' do service.shutdown end
+      service.start
+    end
+    sleep(1) # leave sometime for the previous webrick to teardown
+    Process.detach(@pid)
+  end
+
+  after do
+    Process.kill('INT', @pid)
```

Comment:
You want SIGKILL here, not SIGINT (which may or may not terminate the process in time for the next spec, which can result into flakyness due to ports still being in use).